File: string-calculator/csharp/src/StringCalculator/Calculator.cs
using System.Linq;

namespace StringCalculator;

public static class Calculator
{
    public static int Add(string numbers)
    {
        if (numbers == "") return 0;

        var delimiters = new[] { ",", "\n" };
        var body = numbers;

        if (numbers.StartsWith("//"))
        {
            var headerEnd = numbers.IndexOf('\n');
            var header = numbers.Substring(2, headerEnd - 2);
            delimiters = new[] { header };
            body = numbers.Substring(headerEnd + 1);
        }

        return body.Split(delimiters, System.StringSplitOptions.None).Sum(int.Parse);
    }
}

File: string-calculator/python/src/string_calculator/calculator.py
import re


def add(numbers: str) -> int:
    if numbers == "":
        return 0

    delimiter_pattern = r"[,\n]"
    body = numbers

    if numbers.startswith("//"):
        header_end = numbers.index("\n")
        header = numbers[2:header_end]
        delimiter_pattern = re.escape(header)
        body = numbers[header_end + 1:]

    return sum(int(token) for token in re.split(delimiter_pattern, body))
